### PR TITLE
rgw: add detailed error message for PutACLs

### PR DESCRIPTION
**Pull request by ZVampirEM77**
Keep the error messages same with AWS S3

Now the error messages returned by RGW when PutACLs failed are like
# Invalid ID

> \<?xml version="1.0" encoding="UTF-8"?\>\<Error\>\<Code\>InvalidArgument\</Code\>\<Message\>Invalid id\</Message\>\<BucketName\>111\</BucketName\>\<RequestId\>tx00000000000000000000a-005d7cb275-108b-default\</RequestId\>\<HostId\>108b-default-default\</HostId\>\</Error\>

# Invalid email address

> \<?xml version="1.0" encoding="UTF-8"?\>\<Error\>\<Code\>UnresolvableGrantByEmailAddress\</Code\>\<Message\>The e-mail address you provided does not match any account on record.\</Message\>\<BucketName\>111\</BucketName\>\<RequestId\>tx000000000000000000016-005d7cb337-108b-default\</RequestId\>\<HostId\>108b-default-default\</HostId\>\</Error\>

# Invalid group uri

> \<?xml version="1.0" encoding="UTF-8"?\>\<Error\>\<Code\>InvalidArgument\</Code\>\<Message\>Invalid group uri\</Message\>\<BucketName\>111\</BucketName\>\<RequestId\>tx000000000000000000015-005d7cb2f5-108b-default\</RequestId\>\<HostId\>108b-default-default\</HostId\>\</Error\>`

Signed-off-by: Enming Zhang <zvampirem77@gmail.com>

## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard backend`
- `jenkins test docs`
- `jenkins render docs`

</details>

### CODE REVIEW

File: src/rgw/rgw_op.cc [hunk 1]
```diff
@@ -5339,9 +5339,12 @@ void RGWPutACLs::execute()
     *_dout << dendl;
   }
 
-  op_ret = policy->rebuild(store->ctl()->user, &owner, new_policy);
-  if (op_ret < 0)
+  std::string err_msg;
```
Comment (yuvalif) [inline src/rgw/rgw_op.cc]: you can pass ``s->err.message`` directly into the function. no need for this temp variable.

File: src/rgw/rgw_acl_s3.cc [hunk 2]
```diff
@@ -489,6 +490,7 @@ int RGWAccessControlPolicy_S3::rebuild(RGWUserCtl *user_ctl, ACLOwner *owner, RG
   RGWUserInfo owner_info;
   if (user_ctl->get_info_by_uid(owner->get_id(), &owner_info, null_yield) < 0) {
     ldout(cct, 10) << "owner info does not exist" << dendl;
```
Comment (yuvalif) [inline src/rgw/rgw_acl_s3.cc]: probably better to unify the log message with the error message sent in the reply. put the string into the error message and then pass it also to ldout

Comment (ZVampirEM77) [reply src/rgw/rgw_acl_s3.cc]: I think the log message is used to help developers to find the issues and bugs in code, but the error message returned by http server is used to tell user what happens and why returns an error response.

So, I think it is not necessary to unify the log message with the error message, the log message should contain more detailed information to help developers to solve issues.

File: src/rgw/rgw_op.cc (hunk 1 above)
Comment (ZVampirEM77) [reply src/rgw/rgw_op.cc]: @yuvalif 
done~

File: src/rgw/rgw_acl_s3.cc (hunk 2 above)
Comment (yuvalif) [reply src/rgw/rgw_acl_s3.cc]: > I think the log message is used to help developers to find the issues and bugs in code, but the error message returned by http server is used to tell user what happens and why returns an error response.
> 
> So, I think it is not necessary to unify the log message with the error message, the log message should contain more detailed information to help developers to solve issues.

this is true for the log message in line 536 (you don't want to put function names in the HTTP reply...) but in the other cases, the log and error message are pretty much the same, only differently phrased.

Comment (ZVampirEM77) [reply src/rgw/rgw_acl_s3.cc]: > > I think the log message is used to help developers to find the issues and bugs in code, but the error message returned by http server is used to tell user what happens and why returns an error response.
> > So, I think it is not necessary to unify the log message with the error message, the log message should contain more detailed information to help developers to solve issues.
> 
> this is true for the log message in line 536 (you don't want to put function names in the HTTP reply...) but in the other cases, the log and error message are pretty much the same, only differently phrased.

I think we should keep the response error message same with AWS S3 for the compatibility. What do you think about it? @cbodley

Comment (yuvalif) [reply src/rgw/rgw_acl_s3.cc]: @ZVampirEM77 if this is a matter of compatibility then I agree. More important than internal consistency

**Comment by ZVampirEM77** [comment]
@cbodley ping~

**Pull request merged**